bloom-housing/ui-components
# refactor: remove translations

# Pull Request Template

Issue [#3278-Core](https://github.com/bloom-housing/bloom/issues/3278)

## Description

This PR removes translations from UIC - there is a connected [PR in core](https://github.com/bloom-housing/bloom/pull/3277) that brings the translations back into shared-helpers.

## How Can This Be Tested/Reviewed?

There are a lot of files touched but very few code changes. The only thing this change affects on the UIC side is Storybook where initially stories had a lot of `{{ Missing Translation Phrases }}`.

Currently most of our components do explicitly call `t(...)` and expect a certain string to be present, but they also allow for overrides through a `strings` prop if they do so. The changes here are mostly just using that strings prop versus calling `t("...")` in our stories and our tests.

There were a few components that were missing the ability to pass in some strings, so the only component file changes can be found in `AgTable`, `AgPagination`, `SubmitApplication` and`GetApplication`, and it's just the addition of new string props.

There are two components with issues that remain in Storybook and that is the Select component and our Table components. Both of these components expect a prop to be passed in as a string in the format `t.example` and then we call `t(...)` internally. These components would require a refactor to be able to pass in all strings as props, and render in isolation in this Storybook without any local translation files.

This is going to be a moderate/significant effort for the Table component. I recommend we table (hah) that for now. The result is that the table stories and select stories will render with a few "missing translation" strings in the mean time.

This will require no code changes in core outside of adding the strings files there and adding the right path to the config.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go
